app/services/account_reset/grant_request.rb

```diff
@@ -21,8 +22,21 @@ def call
 
     private
 
-    def account_reset_request
-      AccountResetRequest.create_or_find_by(user_id: @user_id)
+    def fraud_user?(arr)
+      arr.user.fraud_review_pending? ||
+        arr.user.fraud_rejection?
+    end
+
+    def fraud_wait_period_not_met?(arr)
+      if fraud_wait_period_days.present?
+        return arr.requested_at > (Time.zone.now - fraud_wait_period_days.days)
```

Comment:
Related to previous comment, `fraud_wait_period_days` can be `nil`.

```
> nil.days
NoMethodError: undefined method `days' for nil
```

Reply:
thanks for catching! ill add spec for that

Reply:
Looks like it shouldnt occur since fraud_wait_period_days we check if its present. I added a couple nil specs to ensure that this isn't occurring anywhere.